docs/recipes/UsageWithTypescript.md

```diff
@@ -322,109 +362,234 @@ const MyComponent = (props: Props) => (
+  'users/fetchById',
+  // Declare the type your function argument here:
+  async (userId: number) => {
+    const response = await fetch(`https://reqres.in/api/users/${userId}`)
+    // Inferred return type: Promise<MyData>
+    return (await response.json()) as MyData
+  }
+)
+
+// the parameter of `fetchUserById` is automatically inferred to `number` here
+// and dispatching the resulting thunkAction will return a Promise of a correctly
+// typed "fulfilled" or "rejected" action.
+const lastReturnedAction = await store.dispatch(fetchUserById(3))
+```
+
+If you need to modify the types of the `thunkApi` parameter, such as supplying the type of the `state` returned by `getState()`, you must supply the first two generic arguments for return type and payload argument, plus whicher "thunkApi argument fields" are relevant in an object:
+
+```ts
+const fetchUserById = createAsyncThunk<
+  // Return type of the payload creator
+  MyData,
+  // First argument to the payload creator
+  number,
+  {
+    // Optional fields for defining thunkApi field types
+    dispatch: AppDispatch
+    state: State
+    extra: {
+      jwt: string
+    }
+  }
+>('users/fetchById', async (userId, thunkApi) => {
+  const response = await fetch(`https://reqres.in/api/users/${userId}`, {
+    headers: {
+      Authorization: `Bearer ${thunkApi.extra.jwt}`
+    }
+  })
+  return (await response.json()) as MyData
 })
 ```
 
-### Typing `createSlice`
+### Typing `createEntityAdapter`
 
-Similar to `createReducer`, `createSlice` will infer the type of its state value from the `initialState` argument. Action types should be declared explicitly, and will be reused for the generated action creators:
+Typing `createEntityAdapter` only requires you to specify the entity type as the single generic argument. This typically looks like:
 
 ```ts
-const counterSlice = createSlice({
-  name: 'counter',
-  initialState: 0 as number,
+interface Book {
+  bookId: number
+  title: string
+  // ...
+}
+
+// highlight-next-line
+const booksAdapter = createEntityAdapter<Book>({
+  selectId: book => book.bookId,
+  sortComparer: (a, b) => a.title.localeCompare(b.title)
+})
+
+const booksSlice = createSlice({
+  name: 'books',
+  // highlight-start
+  // The type of the state is inferred here
+  initialState: booksAdapter.getInitialState(),
+  // highlight-end
   reducers: {
-    increment(state, action: PayloadAction<number>) {
-      return state + action.payload
+    bookAdded: booksAdapter.addOne,
+    booksReceived(state, action: PayloadAction<{ books: Book[] }>) {
+      booksAdapter.setAll(state, action.payload.books)
     }
   }
 })
 ```
 
+## Additional Recommendations
+
+### Use the React-Redux Hooks API
+
+**We recommend using the React-Redux hooks API as the default approach**. The hooks API is much simpler to use with TypeScript, as `useSelector` is a simple hook that takes a selector function, and the return type is easily inferred from the type of the `state` argument.
```

Comment:
Just a note here, some edits have been set to `React Redux`, while others are ` React-Redux` as it is here. Not a deal, but thought I'd point it out.